sqlglot/parser.py

```diff
@@ -1167,28 +1167,61 @@ def _parse_from(self):
         return self.expression(exp.From, expressions=self._parse_csv(self._parse_table))
 
     def _parse_lateral(self):
-        if not self._match(TokenType.LATERAL):
-            return None
+        index = self._index
+        if self._match_set([TokenType.OUTER, TokenType.CROSS]):
+            cross = self._prev.token_type == TokenType.CROSS
+            kind = "inner" if cross else "outer"
+            side = None if cross else "left"
+            outer = not cross
 
-        subquery = self._parse_select(table=True)
+            if not self._match(TokenType.APPLY):
+                self._retreat(index)
+                return None
 
-        if subquery:
-            return self.expression(exp.Lateral, this=subquery)
+            lat_type = self._prev.token_type
+
+        elif self._match(TokenType.LATERAL):
+            lat_type = self._prev.token_type
+        else:
+            return None
 
-        self._match(TokenType.VIEW)
+        this = self._parse_select(table=True)
+        view = self._match(TokenType.VIEW)
         outer = self._match(TokenType.OUTER)
 
-        return self.expression(
+        if not this:
+            this = self._parse_function()
+
+        table_alias = self._parse_id_var(any_token=False)
+        columns = None
+        if self._match_set([TokenType.ALIAS, TokenType.L_PAREN]):
```

Comment:
i think this could probably get cleaned up

Reply:
What do you mean by this? Would you rewrite this in a different manner?

Reply:
yes, but i don't have time to dig in deeply perhaps @GeorgeSittas can take a look 

maybe tonight

Reply:
Yep, wrapping up some stuff and will take a closer look to see if I can help.

Reply:
Rewrote the part, let me know what you think.

Reply:
Looks good.